pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -2494,8 +2498,16 @@ default 'raise'
             # We can avoid going through pydatetime paths, which is robust
             #  to datetimes outside of pydatetime range.
             ts = _TSObject()
-            check_dts_bounds(&dts, self._creso)
-            ts.value = npy_datetimestruct_to_datetime(self._creso, &dts)
+            try:
+                ts.value = npy_datetimestruct_to_datetime(self._creso, &dts)
+            except OverflowError as err:
+                # TODO: this is copied from check_dts_bounds, with the thought that
+                # eventually we can get rid of check_dts_bounds
+                fmt = (f"{dts.year}-{dts.month:02d}-{dts.day:02d} "
+                       f"{dts.hour:02d}:{dts.min:02d}:{dts.sec:02d}")
+                raise OutOfBoundsDatetime(
+                    f"Out of bounds nanosecond timestamp: {fmt}"
```

Comment:
Not sure how tied we are to this error message. Reconstructed to pass tests for now, but worth noting that the `nanosecond` part of the message is incorrect since we support non-nano resolutions

Reply:
Not at all.  There's a comment in np_datetime.pyx `# TODO: "nanosecond" in the message assumes NPY_FR_ns`